echopop/workflow/feat_hake.py

```diff
@@ -314,103 +314,194 @@
-kriging_params_dict, variogram_params_dict = load_data.load_kriging_variogram_params(
-    geostatistic_params_filepath,
-    geostatistic_params_sheet_name,
-    FEAT_TO_ECHOPOP_GEOSTATS_PARAMS_COLUMNS
+# 
+dict_kriging_params, dict_variogram_params = load_data.load_kriging_variogram_params(
+    geostatistic_params_filepath=(
+        ROOT_PATH / "Kriging_files/default_vario_krig_settings_2019_US_CAN.xlsx"
+    ),
+    sheet_name="Sheet1",
+    column_name_map=FEAT_TO_ECHOPOP_GEOSTATS_PARAMS_COLUMNS
 )
 
-# ===========================================
-# Compute biological composition based on stratum
-length_bins: np.array  # length bin specification
-df_length_weight, df_regression = get_proportions.length_weight_regression(
-    dict_df_bio["specimen"], length_bins
+# ==================================================================================================
+# ==================================================================================================
+# DATA PROCESSING
+# ==================================================================================================
+# Generate binned distributions [age, length]
+# -------------------------------------------
+age_bins: npt.NDArray[np.number] = np.linspace(start=1., stop=22., num=22)
+length_bins: npt.NDArray[np.number] = np.linspace(start=2., stop=80., num=40)
+
+# 
+# ---- Length
+utils.binify(
+    data=dict_df_bio, bins=length_bins, bin_column="length", 
 )
-# df_regression seems unused afterwards -- good as a record?
 
-# Get counts ----------------
-df_aged_counts = get_proportions.fish_count(  # previously "aged_number_distribution"
-    df_specimen=dict_df_bio["specimen"],
-    df_length=dict_df_bio["length"],
-    aged=True,
-    sexed=True,
+# Age
+utils.binify(
+    data=dict_df_bio, bins=age_bins, bin_column="age",
 )
-df_unaged_counts = get_proportions.fish_count(  # previously "unaged_number_distribution"
-    df_specimen=dict_df_bio["specimen"],
-    df_length=dict_df_bio["length"],
-    aged=False,
-    sexed=True,
+
+# ==================================================================================================
+# Fit length-weight regression to the binned data
+# -----------------------------------------------
+
+# Dictionary for length-weight regression coefficients
+dict_length_weight_coefs = {}
+
+# For all fish
+dict_length_weight_coefs["all"] = dict_df_bio["specimen"].assign(sex="all").groupby(["sex"]).apply(
+    biology.fit_length_weight_regression,
+    include_groups=False
 )
-# Previously there was also "aged_number_distribution_filtered"
-# but it is simply df_aged_counts with unsexed fish removed,
-# I think it is better to have that explicitly in the code,
-# so removed OUTSIDE of the get_fish_count function
 
+# Sex-specific
+dict_length_weight_coefs["sex"] = dict_df_bio["specimen"].groupby(["sex"]).apply(
+    biology.fit_length_weight_regression,
+    include_groups=False
+)
```

Comment:
`biology.fit_length_weight_regression()` does not have the argument `include_groups=False`.